common/flatpak-dir.c

```diff
@@ -7627,52 +7627,58 @@ get_appstream_for_ref (FlatpakDir          *self,
+ * If @ref should not be installed, an error is returned. */
+static gboolean
+flatpak_dir_user_check_parental_controls (FlatpakDir    *self,
+                                          const gchar   *remote_name,
+                                          const gchar   *ref,
+                                          GCancellable  *cancellable,
+                                          GError       **error)
 {
-  AsContentRating *rating = NULL;
-
-  rating = as_app_get_content_rating (app, "oars-1.1");
-  if (rating != NULL)
-    return rating;
+  g_autoptr(AsApp) app = NULL;
+  g_autoptr(GError) local_error = NULL;
 
-  rating = as_app_get_content_rating (app, "oars-1.0");
-  if (rating != NULL)
-    return rating;
+  /* The system helper implements its own checks via polkit. */
+  if (!self->user)
+    return TRUE;
 
-  return NULL;
-}
+  /* Get the AppStream data for this app, so we can check its content rating. */
+  app = flatpak_dir_get_appstream_for_ref (self, remote_name, ref, NULL,
+                                           cancellable, &local_error);
 
-/* Check whether the OARS rating in @rating is as, or less, extreme than the
- * user’s preferences in @filter. If so (i.e. if the app is suitable for this
- * user to use), return %TRUE; otherwise return %FALSE.
- *
- * @rating may be %NULL if no OARS ratings are provided for the app. If so,
- * we have to assume the most restrictive ratings. */
-static gboolean
-appstream_check_rating (AsContentRating *rating  /* (nullable) */,
-                        EpcAppFilter    *filter)
-{
-  g_autofree const gchar **oars_sections = epc_app_filter_get_oars_sections (filter);
+  if (local_error != NULL)
+    {
+      g_propagate_error (error, g_steal_pointer (&local_error));
+      return FALSE;
+    }
 
-  for (gsize i = 0; oars_sections[i] != NULL; i++)
+  /* Check that this user is actually allowed to install this app. If we
+   * couldn’t get the AppStream data for it, assume it’s installable by
+   * anyone. */
+  if (app != NULL)
     {
-      AsContentRatingValue rating_value;
-      EpcAppFilterOarsValue filter_value = epc_app_filter_get_oars_value (filter,
-                                                                          oars_sections[i]);
+      g_autoptr(EpcAppFilter) app_filter = NULL;
+      app_filter = epc_get_app_filter (NULL, getuid (), TRUE, cancellable, error);
+      if (app_filter == NULL)
+        return FALSE;
 
-      if (rating != NULL)
-        rating_value = as_content_rating_get_value (rating, oars_sections[i]);
-      else
-        rating_value = AS_CONTENT_RATING_VALUE_INTENSE;
+      AsContentRating *rating = flatpak_appstream_get_latest_content_rating (app);
 
-      if (rating_value == AS_CONTENT_RATING_VALUE_UNKNOWN ||
-          filter_value == EPC_APP_FILTER_OARS_VALUE_UNKNOWN)
-        continue;
-      else if (convert_app_filter_oars_value (filter_value) < rating_value)
-        return FALSE;
+      g_assert (self->user);
```

Comment:
This assertion is redundant with checking this condition a few lines above. I guess it's harmless though.

Reply:
I added it deliberately in case the code got refactored later and the condition from a few lines above was no longer on every control flow to here. Good eyes though.